extensions/amp-list/0.1/amp-list.js

```diff
@@ -446,6 +459,59 @@ export class AmpList extends AMP.BaseElement {
       });
   }
 
+  /**
+   * Gets the json from an amp-script uri.
+   *
+   * @param {string} src
+   * @return {Promise<!JsonObject>}
+   * @private
+   */
+  getAmpScriptJson_(src) {
+    if (!isExperimentOn(this.win, 'protocol-adapters')) {
+      return Promise.reject(`Experiment 'protocol-adapters' is not turned on.`);
+    }
+    return Services.scriptForDocOrNull(this.element)
+      .then((ampScript) => {
+        userAssert(
+          ampScript,
+          '[amp-list]: "amp-script" URLs require amp-script to be installed.'
+        );
+        userAssert(
+          !this.ssrTemplateHelper_.isEnabled(),
+          '[amp-list]: "amp-script" URIs cannot be used in SSR mode.'
+        );
+
+        const args = src.slice(AMP_SCRIPT_URI_SCHEME.length).split('.');
+        userAssert(
+          args.length === 2 && args[0].length > 0 && args[1].length > 0,
+          '[amp-list]: "amp-script" URIs must be of the format "scriptId.functionIdentifier".'
+        );
+
+        const ampScriptId = args[0];
+        const fnIdentifier = args[1];
+        const ampScriptEl = this.element
+          .getAmpDoc()
+          .getRootNode()
+          .querySelector(
+            `amp-script[script=${escapeCssSelectorIdent(ampScriptId)}]`
```

Comment:
as of now, I don't think is necessary to give IDs to amp-script elements. The link between `<amp-script>` and `<script>` is the script= attr.

Reply:
AMP `[on]` actions use reference by ID, so this would be be more consistent and IMO more straightforward.

Also: 
> What if the amp-script has a src attribute instead?

Reply:
Ah. I see what you mean now! I'm convinced / will update accordingly.